### PR TITLE
Added check_on_raw=True for existing tests

### PR DESCRIPTION
**Pull request by aditipande96**
Signed-off-by: Aditi Pande <aditi.pande@seagate.com>

# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by aditipande96** [comment]
======= 2323 tests collected in 11.11s ======

**Comment by aditipande96** [comment]
PASSED tests/csm/rest/test_capacity_quota.py::TestCapacityQuota::test_40632
PASSED tests/csm/rest/test_capacity_quota.py::TestCapacityQuota::test_40633
PASSED tests/csm/rest/test_capacity_quota.py::TestCapacityQuota::test_40634
PASSED tests/csm/rest/test_capacity_quota.py::TestCapacityQuota::test_40635
PASSED tests/csm/rest/test_capacity_quota.py::TestCapacityQuota::test_40637
PASSED tests/csm/rest/test_capacity_quota.py::TestCapacityQuota::test_41127
PASSED tests/csm/rest/test_capacity_quota.py::TestCapacityQuota::test_41746
PASSED tests/csm/rest/test_capacity_quota.py::TestCapacityQuota::test_41750
PASSED tests/csm/rest/test_capacity_quota.py::TestCapacityQuota::test_41151
PASSED tests/csm/rest/test_capacity_quota.py::TestCapacityQuota::test_41128
PASSED tests/csm/rest/test_capacity_quota.py::TestCapacityQuota::test_41966
PASSED tests/csm/rest/test_capacity_quota.py::TestCapacityQuota::test_41967

Blocked due to CORTX-32043:
SKIPPED tests/csm/rest/test_capacity_quota.py::TestCapacityQuota::test_40636
SKIPPED tests/csm/rest/test_capacity_quota.py::TestCapacityQuota::test_41757
SKIPPED tests/csm/rest/test_capacity_quota.py::TestCapacityQuota::test_41758

File: libs/csm/rest/csm_rest_quota.py [hunk 1]
```diff
@@ -64,14 +64,16 @@ def get_user_quota(self, uid, **kwargs):
 
     #pylint disable=no-self-use
     def iam_user_quota_payload(self,enabled: str,
-                        max_size: int, max_objects: int):
+                        max_size: int, max_objects: int,
+                        check_on_raw: str):
```
Comment (kachhwahadivya) [inline libs/csm/rest/csm_rest_quota.py:68]: Can u add details of the parameter in the function comment. Also expected units for max_size?

File: tests/csm/rest/test_capacity_quota.py [hunk 2]
```diff
@@ -355,18 +387,24 @@ def test_40636(self):
...
-                self.obj_name, bucket, akey, skey), "Failed to delete bucket."
+            res, obj_list = self.csm_obj.verify_max_size(max_size, self.akey, self.skey,
+                                     self.bucket)
+            assert True, res[1]
+            for objs in obj_list:
+                try:
+                   self.log.info("Step 3: Delete object")
+                   assert s3_misc.delete_object(
+                       objs, self.bucket, self.akey, self.skey), "Failed to delete object."
+                except:
```
Comment (kachhwahadivya) [inline tests/csm/rest/test_capacity_quota.py:406]: This is very generic exception handling. It should only handle object doesn't exist exception.

File: tests/csm/rest/test_capacity_quota.py [hunk 3]
```diff
@@ -298,18 +324,24 @@ def test_40635(self):
...
-            self.obj_name, self.bucket, self.akey, self.skey), "Failed to delete bucket."
+        res, obj_list = self.csm_obj.verify_max_size(max_size, self.akey, self.skey,
+                                     self.bucket)
+        assert True, res[1]
+        for objs in obj_list:
+             try:
+               self.log.info("Step 3: Delete object")
+               assert s3_misc.delete_object(
+                    objs, self.bucket, self.akey, self.skey), "Failed to delete object."
+             except:
```
Comment (kachhwahadivya) [inline tests/csm/rest/test_capacity_quota.py:343]: This is very generic exception handling. It should only handle object doesn't exist exception.

Comment (kachhwahadivya) [reply tests/csm/rest/test_capacity_quota.py:343]: Also can u help me understand why we need this?

File: tests/csm/rest/test_capacity_quota.py [hunk 4]
```diff
@@ -298,18 +324,24 @@ def test_40635(self):
...
         self.log.info("Response : %s", resp)
         self.log.info("Step 3: Perform max size verification")
-        res = self.csm_obj.verify_max_size(max_size, self.akey, self.skey, self.bucket)
-        assert_utils.assert_false(res[0], res[1])
-        self.log.info("Step 4: Delete object")
-        assert s3_misc.delete_object(
-            self.obj_name, self.bucket, self.akey, self.skey), "Failed to delete bucket."
+        res, obj_list = self.csm_obj.verify_max_size(max_size, self.akey, self.skey,
+                                     self.bucket)
+        assert True, res[1]
```
Comment (kachhwahadivya) [inline tests/csm/rest/test_capacity_quota.py:337]: What is the use of this assert True? Can you please correct this?

File: tests/csm/rest/test_capacity_quota.py [hunk 5]
```diff
@@ -502,27 +538,23 @@ def test_40639(self):
...
         self.log.info("Step 2: Start Multipart upload S3 operations of X Mb")
+        bucket = "iam-user-bucket-" + str(int(time.time_ns()))
+        obj_name = f'{self.obj_name_prefix}{time.perf_counter_ns()}'
         res = self.create_bucket_to_upload_parts(
-            self.bucket,
-            self.obj_name,
-            max_size,
+            bucket,
+            obj_name,
+            int(max_size/1024),
```
Comment (kachhwahadivya) [inline tests/csm/rest/test_capacity_quota.py:550]: Can u help me understand what is the max_size unit assumed throught out these tests and libs?

Comment (aditipande96) [reply tests/csm/rest/test_capacity_quota.py:550]: max size considered is in bytes and block size of 1K or 1M is considered as per test requirement.

File: libs/csm/rest/csm_rest_quota.py (hunk 1 above)
Comment (aditipande96) [reply libs/csm/rest/csm_rest_quota.py:68]: sure

File: tests/csm/rest/test_capacity_quota.py (hunk 2 above)
Comment (aditipande96) [reply tests/csm/rest/test_capacity_quota.py]: removed since it wasn't needed

File: tests/csm/rest/test_capacity_quota.py (hunk 3 above)
Comment (aditipande96) [reply tests/csm/rest/test_capacity_quota.py]: removed since it wasn't needed

File: tests/csm/rest/test_capacity_quota.py (hunk 4 above)
Comment (aditipande96) [reply tests/csm/rest/test_capacity_quota.py]: corrected

**Comment by aditipande96** [comment]
========= 2323 tests collected in 8.94s ========

PASSED tests/csm/rest/test_capacity_quota.py::TestCapacityQuota::test_41968
PASSED tests/csm/rest/test_capacity_quota.py::TestCapacityQuota::test_41965
PASSED tests/csm/rest/test_capacity_quota.py::TestCapacityQuota::test_41954
PASSED tests/csm/rest/test_capacity_quota.py::TestCapacityQuota::test_41172

File: tests/csm/rest/test_capacity_quota.py [hunk 6]
```diff
@@ -96,6 +101,7 @@ def setup_method(self):
         assert_utils.assert_true(resp1[0], resp1[1])
         self.akey = resp.json()["keys"][0]["access_key"]
         self.skey = resp.json()["keys"][0]["secret_key"]
+        self.s3t_obj = S3TestLib(access_key=self.akey, secret_key=self.skey)
         self.bucket = "iam-user-bucket-" + str(int(time.time_ns()))
```
Comment (RajeshPratapSInghChouhan) [inline tests/csm/rest/test_capacity_quota.py:105]: if we are not using it, we can remove

Comment (RajeshPratapSInghChouhan) [reply tests/csm/rest/test_capacity_quota.py:105]: or it is used in some old test ?

Comment (aditipande96) [reply tests/csm/rest/test_capacity_quota.py:105]: It is being used in multipart tests

File: tests/csm/rest/test_iam_users.py [hunk 7]
```diff
@@ -4939,15 +4944,18 @@ def test_42284(self):
...
         self.log.info("Step 3: Send GET request with max_entries as 15 and "
                       "marker as in between user")
         resp = self.csm_obj.list_iam_users_rgw(max_entries=15, marker=marker)
         assert_utils.assert_equals(resp.status_code, HTTPStatus.OK, "Status check failed")
         count_new = resp.json()["count"]
         get_user_list = resp.json()["users"]
-        actual_entries = self.csm_conf["common"]["num_users"] - user_index + 1
-        assert_utils.assert_equals(count_new, actual_entries, "Entries not returned as expected")
+        if len(resp1.json()["users"]) > 11:
+            assert_utils.assert_equals(count_new, 15, "Entries not returned as expected")
```
Comment (RajeshPratapSInghChouhan) [inline tests/csm/rest/test_iam_users.py:4955]: 11 & 15 are constant ?

Comment (RajeshPratapSInghChouhan) [reply tests/csm/rest/test_iam_users.py:4955]: can we have these in test yaml or in other file

Comment (aditipande96) [reply tests/csm/rest/test_iam_users.py:4955]: sure, will move it to test config

File: tests/csm/rest/test_capacity_quota.py [hunk 8]
```diff
@@ -90,12 +95,14 @@ def setup_method(self):
                                  "IAM user creation failed")
         self.uid = payload["uid"]
         self.user_id = resp.json()['tenant'] + "$" + self.uid
-        self.created_iam_users.add(self.user_id)
+        usr_val = resp.json()["keys"][0]
+        self.created_iam_users.update({usr_val['user']:usr_val})
```
Comment (RajeshPratapSInghChouhan) [inline tests/csm/rest/test_capacity_quota.py:99]: please test and revert if this not working.

Comment (aditipande96) [reply tests/csm/rest/test_capacity_quota.py:99]: sure, will remove

File: tests/csm/rest/test_capacity_quota.py [hunk 9]
```diff
@@ -318,7 +342,7 @@ def test_40635(self):
 
 
     # pylint: disable-msg=too-many-locals
-    @pytest.mark.skip(reason="CORTX-32043")
+    #@pytest.mark.skip(reason="CORTX-32043")
```
Comment (RajeshPratapSInghChouhan) [inline tests/csm/rest/test_capacity_quota.py:345]: can remove if not required now, else revert

File: tests/csm/rest/test_capacity_quota.py [hunk 10]
```diff
@@ -798,7 +821,7 @@ def test_41750(self):
         assert_utils.assert_greater_equal(m_size, total_size, "Total Used Size mismatch found ")
         self.log.info("##### Test ended -  %s #####", test_case_name)
 
-
+    #@pytest.mark.skip(reason="CORTX-32043")
```
Comment (RajeshPratapSInghChouhan) [inline tests/csm/rest/test_capacity_quota.py:824]: please remove if not required

File: tests/csm/rest/test_capacity_quota.py [hunk 11]
```diff
@@ -1216,20 +1274,23 @@ def test_41173(self):
...
-            total_size / (1024 * 1024), total_data_size, "Total Size mismatch found")
+            total_size / 1024, total_data_size, "Total Size mismatch found")
         assert_utils.assert_greater_equal(
-            total_size / (1024 * 1024), m_size, "Total Used Size mismatch found ")
+                m_size, total_size, "Total Used Size mismatch found ")
 
         self.log.info("##### Test ended -  %s #####", test_case_name)
 
-
+    #@pytest.mark.skip("Feature not ready")
```
Comment (RajeshPratapSInghChouhan) [inline tests/csm/rest/test_capacity_quota.py:1293]: same as above

**Review by RajeshPratapSInghChouhan — APPROVED** [review]
please take a look at the comments and fix Codacy issues

File: tests/csm/rest/test_iam_users.py (hunk 7 above)
Comment (aditipande96) [reply tests/csm/rest/test_iam_users.py]: done

File: tests/csm/rest/test_capacity_quota.py (hunk 8 above)
Comment (aditipande96) [reply tests/csm/rest/test_capacity_quota.py]: done

File: tests/csm/rest/test_capacity_quota.py (hunk 11 above)
Comment (aditipande96) [reply tests/csm/rest/test_capacity_quota.py]: done

**Review by ravindrachoudhari — APPROVED** [review]

**Review by kachhwahadivya — APPROVED** [review]

**Pull request merged**